cmd/test-auth/main.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/guardian/cq-source-github-languages/internal/github"
+)
+
+// sanitizePrivateKey ensures the private key is properly formatted with newlines
+// and handles base64 encoded keys
+func sanitizePrivateKey(key string) ([]byte, error) {
+	// Trim any extra whitespace
+	key = strings.TrimSpace(key)
+
+	// Check if the key is base64 encoded (doesn't start with PEM headers)
+	if !strings.HasPrefix(key, "-----BEGIN") {
+		// Try to decode as base64
+		fmt.Println("Key doesn't start with PEM header, attempting base64 decode...")
+		decoded, err := base64.StdEncoding.DecodeString(key)
+		if err != nil {
+			return nil, fmt.Errorf("key doesn't appear to be in PEM format and failed base64 decoding: %w", err)
+		}
+
+		// Check if decoded content looks like a PEM key
+		decodedStr := string(decoded)
+		if strings.Contains(decodedStr, "-----BEGIN") {
+			fmt.Println("Successfully decoded base64 key to PEM format")
+			key = decodedStr
+		} else {
+			return nil, fmt.Errorf("base64 decoded content doesn't appear to be a valid PEM key")
+		}
+	}
+
+	// Replace literal "\n" with actual newlines if they exist
+	key = strings.ReplaceAll(key, "\\n", "\n")
+
+	// Ensure key has proper BEGIN and END markers with newlines
+	if !strings.Contains(key, "-----BEGIN") {
+		fmt.Println("Warning: Private key does not contain BEGIN marker")
+	}
+	if !strings.Contains(key, "-----END") {
+		fmt.Println("Warning: Private key does not contain END marker")
+	}
+
+	return []byte(key), nil
+}
+
+func main() {
+	// Define flags for command line arguments
+	appIDStr := flag.String("app-id", os.Getenv("GITHUB_APP_ID"), "GitHub App ID (or set GITHUB_APP_ID env)")
```

Comment:
minor: How will this behave if a user passes in both env vars and cli args? Which set will be preferred? Could this be a comment?